llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -881,6 +881,9 @@ class VPInstruction : public VPRecipeWithIRFlags,
     // Extracts the first active lane of a vector, where the first operand is
     // the predicate, and the second operand is the vector to extract.
     ExtractFirstActive,
+    // Creates a step vector starting from 0 with a step of 1. The first operand
```

Comment:
Possible alternative - add two arguments for step and start, and use the type of those to drive the typing decision?  This would have it slightly out of sync with the signature of the intrinsic though, so not clear this is a net win.

Reply:
We would need to pass in the potentially truncated step and start which might get tangled up a bit, since they're currently truncated inside `VPWidenIntOrFpInductionRecipe::execute`, i.e. in LLVM IR not VPlan. I think we could refactor this and make VPInstruction::StepVector more powerful (i.e. handle arbitrary steps + starts) after #118638, because the truncated step + start will have been modelled in VPlan by then.